extensions/ql-vscode/src/view/common/FileCodeSnippet/CodeSnippetMessage.tsx

```diff
@@ -65,6 +66,7 @@ export const CodeSnippetMessage = ({
               return (
                 <LocationLink
                   key={index}
+                  onClick={() => sendTelemetry("alert-message-link")}
```

Comment:
Can we extract this function to be outside of the component? By having it here, the `onClick` value will change on every re-render, which could result in a much larger part of the tree being re-rendered. I don't think that would happen here, but I think it's good practice to always keep that in mind.

There are some other files where you use this same pattern, I think those can also be extracted.